enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+  - The number of ManagedClusterSets might be very large;
+  - The new added ManagedClusterSets cannot be bound to the namespace automatically;
+
+### Workflows
+
+#### Workflow 1: Placement Reconciliation
+1. If the placement has old placementdicisions, fetch them all and get a slice of old matched clusters;
+2. Evaluate each cluster predicates (see Workflow 2 below for more details) and get the union of all predicates result;
+3. Enforce the affinity constrains on the union and remove clusters if necessary;
+4. If the placement has no old placementdicisions yet, create placementdicisions;
+5. Otherwise, update the existing placementdicisions, and add/remove placementdicisions if necessary.
+
+#### Workflow 2: Evaluate a cluster predicate
+1. If `ClusterName` is specified, return cluster if it is exists;
+2. If `ClusterSets` is not empty, get clusters belongs to intersection of `ClusterSets` and the ManagedClusterSets bound to the placement namespace;
+3. Otherwise, get clusters from the ManagedClusterSets bound to the placement namespace;
+4. If `RequiredDuringSchedulingRequiredDuringExecution` is specified, filter clusters with label/claim selectors; 
+5. Enforce the affinity constrains on clusters if exist;
+6. If `NumberOfClusters` is not specified, return all matched clusters;
+7. Otherwise, get the intersection of old matched clusters and new matched clusters;
+8. If `NumberOfClusters` is equal to the size of the intersection, return clusters in the intersection;
+9. if `NumberOfClusters` is less than the size of the intersection, select random clusters from the intersection and return;
+10. Otherwise, return clusters in the intersection with some random additional clusters from the remaining of the new matched clusters;
+
+**Note:** The old matched clusters are preferred during the predicate evaluation, which makes the placement decisions as stable as possible.
+
+### Open Questions
+
+1. The API group of the new placement APIs could be `cluster.open-cluster-management.io` or `scheduling.open-cluster-management.io`. Which one is better?
+
+### Test Plan
+
+- Unit tests will cover the functionality of the controllers.
+- Unit tests will cover the new APIs.
+- Integration tests will cover all user stories.
+- And e2e tests will cover the following cases.
+  - Create/update/delete a placement;
+  - ManagedCluster labels/claims changes;
+  - A ManagedClusterSet is bound/unbound to/from the placement namespace;
+  - A ManagedCluster is added/removed to/from a bound ManagedClusterSet;
+
+### Graduation Criteria
+
+#### Alpha
+1. The new APIs is reviewed and accepted;
+2. Implementation is completed to support the functionalities;
+3. Develop test cases to demonstrate that the above user stories work correctly;
+
+#### Beta
+1. The new APIs is able to cover all the capabilities of the current placementrule APIs;
+2. Conversion webhook is provided to migrate existing placement objects to the new placement APIs;
```

Comment:
cc @deads2k is this more appropriate for an offline migration? I don't see this as the job of a conversion webhook

Reply:
> cc @deads2k is this more appropriate for an offline migration? I don't see this as the job of a conversion webhook

I think this may be slightly difficult to parse, but a valid statement.  When going to beta, the existing placement objects will be v1alpha1.  Assuming the API doesn't change dramatically, it should be possible to build a conversion webhook to cleanly migrate from v1alpha1 to v1beta1.

I think they called out the requirement here because kube's general promotion rules do not *require* this style of conversion. The default rules allow for an alpha stage that must be removed before installing a beta.  Though we don't usually do that, it has (rarely) happened that features cannot be auto-promoted.

Reply:
> 2. Conversion webhook is provided to migrate existing placement objects to the new placement APIs;

I removed this item because
- Conversion webhook does not work across API groups;
- For the new placement APIs, upgrading from alpha to beta may not need a webhook to do the conversion;

I agree with @pmorie that an offline migration is needed between the old placementrule APIs and the new placement APIs